File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.py
def solution(n,a,b):
    return matching(a,b,1)

def matching(a,b,count):
    if abs(a - b) == 1 and min(a, b) % 2 == 1:
        return count
    a = (a + 1) // 2
    b = (b + 1) // 2
    count += 1
    return matching(a,b,count)

File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.java
class Solution {
    public int solution(int n, int a, int b) {
        return match(a, b, 1);
    }

    public int match(int a, int b, int count) {
        if (Math.abs(a - b) == 1) return count;
        a = (a + 1) / 2;
        b = (b + 1) / 2;
        count++;
        return match(a, b, count);
    }
}